lua/opencode/config_file.lua

```diff
@@ -1,64 +1,71 @@
+local Promise = require('opencode.promise')
 local M = {
   config_promise = nil,
   project_promise = nil,
   providers_promise = nil,
 }
 
----@return OpencodeConfigFile|nil
-function M.get_opencode_config()
+---@type fun(): Promise<OpencodeConfigFile|nil>
+M.get_opencode_config = Promise.async(function()
   if not M.config_promise then
     local state = require('opencode.state')
     M.config_promise = state.api_client:get_config()
   end
   local ok, result = pcall(function()
-    return M.config_promise:wait()
+    return M.config_promise:await()
   end)
 
   if not ok then
     vim.notify('Error fetching Opencode config: ' .. vim.inspect(result), vim.log.levels.ERROR)
     return nil
   end
 
-  return result --[[@as OpencodeConfigFile|nil]]
-end
+  return result
+end)
 
----@return OpencodeProject|nil
-function M.get_opencode_project()
+---@type fun(): Promise<OpencodeProject|nil>
+M.get_opencode_project = Promise.async(function()
   if not M.project_promise then
     local state = require('opencode.state')
     M.project_promise = state.api_client:get_current_project()
   end
   local ok, result = pcall(function()
-    return M.project_promise:wait()
+    return M.project_promise:await()
   end)
   if not ok then
     vim.notify('Error fetching Opencode project: ' .. vim.inspect(result), vim.log.levels.ERROR)
     return nil
   end
 
   return result --[[@as OpencodeProject|nil]]
-end
+end)
+
+---Get the snapshot storage path for the current workspace
+---@type fun(): Promise<string>
+M.get_workspace_snapshot_path = Promise.async(function()
+  local project = M.get_opencode_project():await() --[[@as OpencodeProject|nil]]
+  if not project then
+    return ''
+  end
+  local home = vim.uv.os_homedir()
+  return home .. '/.local/share/opencode/snapshot/' .. project.id
+end)
 
----@return OpencodeProvidersResponse|nil
+---@return Promise<OpencodeProvidersResponse|nil>
 function M.get_opencode_providers()
   if not M.providers_promise then
     local state = require('opencode.state')
     M.providers_promise = state.api_client:list_providers()
   end
-  local ok, result = pcall(function()
-    return M.providers_promise:wait()
-  end)
-  if not ok then
-    vim.notify('Error fetching Opencode providers: ' .. vim.inspect(result), vim.log.levels.ERROR)
+  return M.providers_promise:catch(function(err)
+    vim.notify('Error fetching Opencode providers: ' .. vim.inspect(err), vim.log.levels.ERROR)
     return nil
-  end
-
-  return result --[[@as OpencodeProvidersResponse|nil]]
+  end)
```

Comment:
since it now returns a promise, might be cleaner to not have it return nil, but something like `Promise.new():resolve(nil)`. Similar with above.